fluent-syntax/src/parser.js

```diff
@@ -566,7 +566,10 @@ export default class FluentParser {
 
       ps.expectChar(')');
 
-      return new AST.CallExpression(literal.id, args);
+      return new AST.CallExpression(
+        new AST.Function(literal.id.name),
```

Comment:
Can you add an assertion here to make sure `literal.id.name` conforms to the EBNF? It should only accept [`[A-Z_?-]+`](https://github.com/projectfluent/fluent/blob/a9fb740ecbf9326a484b812800f62a338a43f7da/spec/fluent.ebnf#L28).